File: step 2/Q5 Best language for regex/regex_speed_test.pl
# regex_speed_test.pl

use strict;
use warnings;
use Time::HiRes qw(time);

my $start_time = time;

open(my $file, "<", "emails.txt") or die "Could not open file: $!";
my $content = do { local $/; <$file> };
close($file);

my $pattern = qr/[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}/;
my @matches = $content =~ /$pattern/g;

my $end_time = time;
my $elapsed_time = $end_time - $start_time;

print "Found " . scalar(@matches) . " email addresses\n";
print "Elapsed time: $elapsed_time seconds\n";

File: step 2/Q5 Best language for regex/regex_speed_test.py
import re
import time

# Les innholdet i emails.txt
with open('emails.txt', 'r') as file:
    content = file.read()

# Definer det regulære uttrykket for å matche e-postadresser
pattern = r'[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}'

# Mål tiden det tar å finne alle matchene
start_time = time.time()
matches = re.findall(pattern, content)
end_time = time.time()

elapsed_time = end_time - start_time

print(f"Found {len(matches)} email addresses")
print(f"Elapsed time: {elapsed_time} seconds")

